test/e2e/specs/editor/various/preview.spec.js

```diff
@@ -0,0 +1,362 @@
+
+		// Save draft and open the preview page right after.
+		await editorPage.click( 'role=button[name="Save draft"i]' );
+		await editorPage.waitForSelector(
+			'role=button[name="Dismiss this notice"] >> text=Draft saved'
+		);
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual( 'aaaaabbbbb' );
+
+		await previewPage.close();
+	} );
+
+	// Verify correct preview. See: https://github.com/WordPress/gutenberg/issues/33616
+	test( 'should display the correct preview when switching between published and draft statuses', async ( {
+		editor,
+		page,
+		previewUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Type Lorem in the title field.
+		await editorPage.type( '[aria-label="Add title"]', 'Lorem' );
+
+		// Open the preview page.
+		const previewPage = await editor.openPreviewPage( editorPage );
+		await previewPage.waitForSelector( 'role=heading[level=1]' );
+
+		// Title in preview should match input.
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		expect( await previewTitle.textContent() ).toEqual( 'Lorem' );
+
+		// Return to editor and publish post.
+		await editorPage.bringToFront();
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.locator( 'role=button[name="Close panel"]' ).click();
+
+		// Change the title and preview again.
+		await editorPage.type( '[aria-label="Add title"]', ' Ipsum' );
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual( 'Lorem Ipsum' );
+
+		// Return to editor and switch to Draft.
+		await editorPage.bringToFront();
+		await page.locator( 'role=button[name="Switch to draft"]' ).click();
+
+		// Change the title.
+		await editorPage.type( '[aria-label="Add title"]', ' Draft' );
+
+		// Open the preview page.
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		expect( await previewTitle.textContent() ).toEqual(
+			'Lorem Ipsum Draft'
+		);
+
+		await previewPage.close();
+	} );
+} );
+
+test.describe( 'Preview with Custom Fields enabled', () => {
+	test.beforeEach( async ( { admin, previewUtils } ) => {
+		await admin.createNewPost();
+		await previewUtils.toggleCustomFieldsOption( true );
+	} );
+
+	test.afterEach( async ( { previewUtils } ) => {
+		await previewUtils.toggleCustomFieldsOption( false );
+	} );
+
+	// Catch regressions of https://github.com/WordPress/gutenberg/issues/12617
+	test( 'displays edits to the post title and content in the preview', async ( {
+		editor,
+		page,
+		previewUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Add an initial title and content.
+		await editorPage.type( 'role=textbox[name="Add title"i]', 'title 1' );
+		await editorPage.type(
+			'role=button[name="Add default block"i]',
+			'content 1'
+		);
```

Comment:
We can also use `insertBlock` to programmatically add content.

```suggestion
		await editor.insertBlock( {
			name: 'core/paragraph',
			attributes: { content: 'content 1' },
		} );
```